backend/fms_core/viewsets/_utils.py

```diff
@@ -141,14 +126,16 @@ def template_submit(self, request):
         if error:
             return HttpResponseBadRequest(json.dumps({"detail": action_data}), content_type="application/json")
 
-        action_def, dataset = action_data
+        action_def, file = action_data
 
-        resource_instance = action_def["resource"]()
-        result = resource_instance.import_data(dataset)
+        importer_instance = action_def["importer"]()
 
-        if result.has_errors() or result.has_validation_errors():
-            # TODO: Better message
-            return HttpResponseBadRequest(json.dumps({"detail": "Template errors encountered in submission"}),
-                                          content_type="application/json")
+        try:
+            importer_instance.import_template(file=file, format='xlsx', dry_run=False)
```

Comment:
What is the plan for csv handling ?

Reply:
Will be added with tests (If even tests are still in csv)